src/toil/jobStores/abstractJobStore.py

```diff
@@ -132,6 +132,44 @@ class AbstractJobStore(object):
     """
     __metaclass__ = ABCMeta
 
+    class Locator(object):
+        @classmethod
+        def parse(cls, s):
+            try:
+                jobStoreName, rest = s.split(':', 1)
+            except ValueError:
+                if s[0] in './':
```

Comment:
Nope, there can be colons in the path. This check need to made before you attempt to split the string. 

Why try to fix what's not broken?